MyAssignment2.Rmd

```diff
@@ -0,0 +1,174 @@
+names_class1<-select(class_2018, name, Q1)
+
+names_class2<-select(class_2018, name, Q3)
+names_class2<-names_class2[-c(1,39),]
+
+names_class3<-select(class_2018,name, Q4)
+names_class3<-arrange(names_class3, Q4)
+names_class3<-names_class3[-c(1:7),]
+
+names_class4<-select(class_2018,name, Q5)
+names_class4<-arrange(names_class4, Q5)
+names_class4<-names_class4[-c(1:12),]
+
+names_class5<-select(class_2018, name, Q6)
+names_class5<-arrange(names_class5, Q6)
+names_class5<-names_class5[-c(1:30),]
+
+names(names_class1)<-c("name","class")
+names(names_class2)<-c("name","class")
+names(names_class3)<-c("name","class")
+names(names_class4)<-c("name","class")
+names(names_class5)<-c("name","class")
+
+n_cl_total<-dplyr::bind_rows(names_class1,names_class2,names_class3, names_class4, names_class5)
+n_cl_total<-arrange(n_cl_total, name)
+n_cl_total$enroll<-1
+#put a space in the class values where there is no space
+n_cl_total$class<-gsub('([A-Z])([0-9])', '\\1 \\2', n_cl_total$class)
+
+
+#create dataframe with each student as an observation and classes as each column
+n2<-spread(n_cl_total,"class", enroll)
+n2[is.na(n2)]<-0
+
+
+#multiply matrix by transpose to get people to people matrix
+n2_matrix<-as.matrix(n2, rownames.force = NA, dimnames(n2$name,class_list))
+n3_matrix<-n2_matrix[,-1]
+mode(n3_matrix)<-"numeric"
+n3_tran<-n3_matrix %*% t(n3_matrix)
+
+#giving names to the matrix dimensions
+rownames(n3_tran)<-n2$name
+colnames(n3_tran)<-n2$name
+n4_tran<-n3_tran
+
+#removing the HUDK4050 data since everyone is connected to each other through it
+noHUDK4050mat<-n4_tran-1
+diag(noHUDK4050mat)<-0
+class_net<-graph.adjacency(noHUDK4050mat, mode="undirected", diag=FALSE)
+plot(class_net, layout=layout_in_circle)
+
+#betweenness calculation without HUDK4050 connections
+B1<-betweenness(class_net, v = V(class_net), directed = FALSE, weights = NULL, normalized = FALSE)
+
+#eigen centrality without HUDK 4050 connections
+E1<-eigen_centrality(class_net, directed = FALSE, scale = TRUE, weights = NULL,
+  options = arpack_defaults)
+
+#degree without HUDK 4050 connections
+Deg1<-degree(class_net, v = V(class_net), mode = c("all", "out", "in", "total"),loops = TRUE, normalized = FALSE)
+
+```
+
+
+Graph and calculations with HUDK 4050 connections included
+
+
+```{r}
+
+##graph with HUDK 4050 connections
+n3_temp<- n3_tran
+diag(n3_temp)<-0
+class_net2<-graph.adjacency(n3_temp, mode="undirected", diag=FALSE)
+plot(class_net2, layout=layout_in_circle)
+
+#betweenness calculation WITH HUDK4050 connections
+B2<-betweenness(class_net2, v = V(class_net2), directed = FALSE, weights = NULL, normalized = FALSE)
+
+#eigen centrality WITH HUDK 4050 connections
+E2<-eigen_centrality(class_net2, directed = FALSE, scale = TRUE, weights = NULL,
+  options = arpack_defaults)
+
+#degree wWITH HUDK 4050 connections
+Deg2<-degree(class_net2, v = V(class_net2), mode = c("all", "out", "in", "total"),loops = TRUE, normalized = FALSE)
+
```

Comment:
Would you expect these to be different? Does this make a difference to who is the most central member of the network?